src/algorithm/intersects.rs

```diff
@@ -448,12 +448,28 @@ mod test {
         let line2 = Line::new(Point::new(0., 6.), Point::new(1.5, 4.5));
         // point on line
         let line3 = Line::new(Point::new(0., 6.), Point::new(3., 3.));
+        // point above line with positive slope
+        let line4 = Line::new(Point::new(1., 2.), Point::new(5., 3.));
+        // point below line with positive slope
+        let line5 = Line::new(Point::new(1., 5.), Point::new(5., 6.));
+        // point above line with negative slope
+        let line6 = Line::new(Point::new(1., 2.), Point::new(5., -3.));
+        // point below line with negative slope
+        let line7 = Line::new(Point::new(1., 6.), Point::new(5., 5.));
         assert!(line1.intersects(&p0));
         assert!(p0.intersects(&line1));
         assert!(!line2.intersects(&p0));
         assert!(!p0.intersects(&line2));
         assert!(line3.intersects(&p0));
         assert!(p0.intersects(&line3));
+        assert!(!line4.intersects(&p0));
+        assert!(!p0.intersects(&line4));
+        assert!(!line5.intersects(&p0));
+        assert!(!p0.intersects(&line5));
+        assert!(!line6.intersects(&p0));
+        assert!(!p0.intersects(&line6));
+        assert!(!line7.intersects(&p0));
+        assert!(!p0.intersects(&line7));
```

Comment:
@michaelkirk: to follow up on your previous comment, the failure occurs when the point is above a line with positive slope or below a line with negative slope. So I included examples above and below lines with positive and negative slopes. On master, lines 465-466 (`line4`) and lines 471-472 (`line7`) will fail.